File: python/leetcode/141. Linked List Cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    def hasCycle(self, head: ListNode) -> bool:
        if head is None:
            return False
        fast = head.next

        while fast:
            if fast == head:
                return True
            fast = fast.next.next if fast.next else None
            head = head.next
        return False

File: go/leetcode/141. Linked List Cycle.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func hasCycle(head *ListNode) bool {
    if head == nil {return false}
    fast := head.Next
    
    for fast != nil {
        if fast == head {return true} 
        if fast.Next != nil {
            fast = fast.Next.Next
        } else {
            fast = nil
        }
        head = head.Next        
    }
        
    return false
}
  